deploy/runtools/switch_model_config.py

```diff
@@ -168,17 +168,17 @@ def local_logged(command: str) -> None:
         local_logged("cd " + switchbuilddir + " && make")
         local_logged("mv " + switchbuilddir + "switch " + switchbuilddir + binaryname)
 
-    def run_switch_simulation_command(self) -> str:
+    def get_switch_simulation_command(self, sudo: bool) -> str:
         """ Return the command to boot the switch."""
         switchlatency = self.fsimswitchnode.switch_switching_latency
         linklatency = self.fsimswitchnode.switch_link_latency
         bandwidth = self.fsimswitchnode.switch_bandwidth
         # insert gdb -ex run --args between sudo and ./ below to start switches in gdb
-        return """screen -S {} -d -m bash -c "script -f -c 'sudo ./{} {} {} {}' switchlog"; sleep 1""".format(self.switch_binary_name(), self.switch_binary_name(), linklatency, switchlatency, bandwidth)
+        return """screen -S {} -d -m bash -c "script -f -c '{} ./{} {} {} {}' switchlog"; sleep 1""".format("sudo" if sudo else "", self.switch_binary_name(), self.switch_binary_name(), linklatency, switchlatency, bandwidth)
 
     def kill_switch_simulation_command(self) -> str:
         """ Return the command to kill the switch. """
-        return """sudo pkill {}""".format(self.switch_binary_name())
+        return """pkill {}""".format(self.switch_binary_name())
```

Comment:
same, this won't work for switches started with sudo

Reply:
Ditto